plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java

```diff
@@ -1860,6 +1861,53 @@ public void testMissingIndex()
         assertTableDoesNotExist("nonexistent_table");
     }
 
+    @Test
+    public void testQueryTableFunction()
+    {
+        // select single record
+        assertQuery("SELECT json_query(result, 'lax $[0][0].hits.hits._source') " +
+                        "FROM TABLE(elasticsearch.system.query(" +
+                        "schema => 'tpch', " +
+                        "index => 'nation', " +
+                        "query => '{\"query\": {\"match\": {\"name\": \"ALGERIA\"}}}')) t(result)",
+                "VALUES '{\"nationkey\":0,\"name\":\"ALGERIA\",\"regionkey\":0,\"comment\":\" haggle. carefully final deposits detect slyly agai\"}'");
```

Comment:
due to the fact that the function seems to dump ES result, i would consider calling a `raw_query`
(especially so that we keep `query` name free for future)

Reply:
That's a good point. In ES parlance, queries are "search queries" (https://www.elastic.co/guide/en/elasticsearch/reference/current/search-your-data.html), and there's a Query DSL (https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html) to formulate them (what we support here). They also have two other languages:
* EQL, for querying log and event data (https://www.elastic.co/guide/en/elasticsearch/reference/current/eql.html)
* SQL (https://www.elastic.co/guide/en/elasticsearch/reference/current/sql-overview.html)